src/downshift.js

```diff
@@ -827,7 +840,8 @@ class Downshift extends Component {
         if (
           (event.target === this._rootNode ||
             !this._rootNode.contains(event.target)) &&
-          this.getState().isOpen
+          this.getState().isOpen &&
+          !this.isInputFocused
```

Comment:
What if instead we just change this to:

```javascript
// above:
const {document} = this.props.environment
// this line
(this.inputId && document.getElementById(this.inputId) === document.activeElement)
```

Then I think we could remove all the focus logic.

Want to give that a try? The tests shouldn't need to change if it works.

Reply:
I like it, and looks like it's working.
It is breaking the test though as it seems like enzyme doesn't really support the `document.activeElement` https://github.com/airbnb/enzyme/issues/1128
I believe I can just go ahead and set the `document.activeElement` in my test.
And what if instead of looking for the `getElementById` we would do as simple text check : 
`(!this.inputId || this.inputId !== document.activeElement.id)` ?

Reply:
Well seems like the testing might get a bit hairier with the `document.activeElement` solution.
To use `document.activeElement` from what I read we might need to use `JSDOM` to simulate this.

Reply:
> And what if instead of looking for the getElementById we would do as simple text check:
 `(!this.inputId || this.inputId !== document.activeElement.id)`

That's fantastic.

Yeah, go ahead and set `document.activeElement` manually in your tests. Kinda weird, but should be ok. If you want you could try to add a browser test via cypress:

https://github.com/paypal/downshift/blob/1dec961ad5e787ad29556c85afe8ddc8bb05968f/cypress/integration/basic.js

Learn more about cypress [here](https://cypress.io).

I think we should have both, one for the quick feedback loop and the other to ensure it works without jsdom

Reply:
If you can't get it to work with the jest tests then that's fine. Add an [`/* istanbul ignore next */` ](https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md) comment on the line above this condition and it should prevent coverage being recorded for that line and we'll cover it using cypress :+1:

Reply:
Well I went with the latest and removed the jest test.
Code coverage is still 100% as this line is in a condition that I believe is considered executed.
I have only heard about cypress but will try to go ahead and make the test

Reply:
Awesome :+1:

Let me know if you need a hand :)

Reply:
Oh, and to start up cypress, run `npm run test:cypress:dev`. I should probably add that to the contributing docs... Would you like to?

Reply:
Hmmm not so sure I have the time atm, but I might do it later if you want.
I believe the `cypress` test is up, it's pretty cool and quite straightforward.
I tested with `npm run test:cypress` though, hope it's equivalent.

Reply:
hi guys! how did u set up the document.activeelement in enzyme? im using activeElement in my component code. However, in my enzyme testing, it won't recognize the activeelement. i kept getting undefined.

Reply:
If you read the thread you commented on you'll see that we didn't do it either....